tests/line_count_analyzer_test.py

```diff
@@ -0,0 +1,11 @@
+import io
+
+from alogamous import analyzer, line_count_analyzer
+
+
+def test_line_count():
+    in_stream = io.StringIO("""line1
+        line2""")
+    out_stream = io.StringIO()
+    analyzer.analyze_log_stream([line_count_analyzer.LineCountAnalyzer()], in_stream, out_stream)
```

Comment:
While you can test this using `analyzer.analyze_log_stream`, I think testing it more directly by calling `read_log_line` and `report` makes for a more informative test.

Specifically, if the behavior of `analyze_log_stream` were to change in some way, we'd only want the test for `analyze_log_stream` to fail, not the test for `LineCountAnalyzer` as well.

Reply:
This is turning out to be more complicated than I would have thought

My first thought was to just copy what the `analyzer.analyze_log_stream` method did, which was `for line in in_stream`, but in the loop run `read_log_line()` on my `LineCountAnalyzer` object. However, since this is now the sub-class and not the ABC class, the line argument is not needed. This means the loop variable `line` is not used, and unfortunately this causes the commit to fail.

Another idea would be to just use the `read_log_line()` method on my analyzer object x number of times, then check that the `line_counter` variable in my analyzer class initializer is x and that the `report()` method writes "number of log lines: x" to the out_stream. However, I think this might be a rather confusing test to look at, as viewers would need to look at the `read_log_line()` method in my analyzer to actually understand why this is properly testing it.

My third idea would be to use a while loop and say something like "while index < Len(in_stream): run 'read_log_line()' on analyzer object". Yet as far as I can tell StringIO objects don't have a way to get the length or search by newline characters or anything, and this requires a seemingly random index variable. To make this work I would need to do some complicated process of reading the lines to a list, or just starting with a list, and its potentially becoming even less readable than the second option. What should I do?

Reply:
I think your second option is probably the best one here. A test like:
```
analyzer.read_log_line("line1")
analyzer.read_log_line("line2")
analyzer.read_log_line("line3")
analyzer.report(out_stream)
assert out_stream.getvalue() == "Saw 3 lines"
```

While it's maybe a little weird, if you do this, you don't need an `in_stream` at all, since the test is the one calling `read_log_line`.